src/ts/fs.d.ts

```diff
@@ -0,0 +1,379 @@
+         */
+        appendFileSync(file: NjsStringLike, data: NjsStringLike | Buffer, options?: WriteFileOptions): void;
+
+        /**
+         * Synchronously creates a directory at the specified `path`.
+         *
+         * @since 0.4.2
+         * @param path A path to a file.
+         * @param options The file mode (or an object specifying the file mode). Defaults to `0o777`.
+         */
+        mkdirSync(path: NjsStringLike, options?: { mode?: number } | number): void;
+
+        /**
+         * Synchronously reads the contents of a directory at the specified `path`.
+         *
+         * @since 0.4.2
+         * @param path A path to a file.
+         * @param options A string that specifies encoding or an object optionally specifying
+         *   the following keys:
+         *   - `encoding` - `'utf8'` (default) or `'buffer'` (since 0.4.4)
+         *   - `withFileTypes` - if set to `true`, the files array will contain `fs.Dirent` objects;
+         *     defaults to `false`.
+         */
+        readdirSync(path: NjsStringLike, options?: { encoding?: "utf8"; withFileTypes?: false; } | "utf8"): NjsStringLike[];
+        readdirSync(path: NjsStringLike, options: { encoding: "buffer"; withFileTypes?: false; } | "buffer"): Buffer[];
+        readdirSync(path: NjsStringLike, options: { encoding?: "utf8" | "buffer"; withFileTypes: true; }): Dirent[];
+
+        /**
+         * Synchronously returns the contents of the file with provided `filename`.
+         * If an encoding is specified, a `string` is returned, otherwise, a `Buffer`.
+         *
+         * @example
+         *   import fs from 'fs'
+         *   var file = fs.readFileSync('/file/path.tar.gz')
+         *   var gzipped = file.slice(0,2).toString('hex') === '1f8b'; gzipped  // => true
+         *
+         * @param path A path to a file.
+         * @param options A string that specifies encoding or an object with the following optional keys:
+         *   - `encoding` - `'utf8'`, `'hex'`, `'base64'`, or `'base64url'` (the last three since 0.4.4).
+         *   - `flag` - file system flag, defaults to `r`.
+         */
+        readFileSync(path: NjsStringLike): Buffer;
+        readFileSync(path: NjsStringLike, options?: { flag?: OpenMode; }): Buffer;
+        readFileSync(path: NjsStringLike, options: { encoding?: FileEncoding; flag?: OpenMode; } | FileEncoding): NjsStringLike;
+
+        /**
+         * Synchronously computes the canonical pathname by resolving `.`, `..` and symbolic links using
+         * `realpath(3)`.
+         *
+         * @since 0.3.9
+         * @param path A path to a file.
+         * @param options The encoding (or an object specifying the encoding), used as the encoding of the result.
+         */
+        realpathSync(path: NjsStringLike, options?: { encoding?: "utf8" } | "utf8"): NjsStringLike;
+        realpathSync(path: NjsStringLike, options: { encoding: "buffer" } | "buffer"): Buffer;  // TODO: is this correct?
```

Comment:
Is this correct?

Reply:
realpathSync(path: NjsStringLike, options?: { encoding?: "utf8" } | "utf8"): string;
realpathSync(path: NjsStringLike, options: { encoding: "buffer" } | "buffer"): Buffer; // this is correct